core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java

```diff
@@ -519,48 +517,53 @@ public boolean isFilterElementTitleBlocksEnabled(DDiagramElement element) {
   }
 
   /**
-   * this function handles the dangling Element Title Blocks in two situations: if the element associated to an element
-   * TB was deleted from model -> delete the TB; if the element associate to an element TB was removed from diagram ->
-   * hide the TB
+   * Handles the dangling Element Title Blocks in two situations: if the element associated to an element TB was deleted
+   * from model -> delete the TB; if the element associate to an element TB was removed from diagram -> hide the TB
    * 
-   * @param list:
-   *          the list of element TB to check
    * @param diagram
-   * @return list of Title Blocks (both Diagram or Element TB) that will be displayed in diagram
+   *          the diagram.
    */
-  private void handleDanglingElementTitleBlocks(List<DAnnotation> list, DDiagram diagram) {
-    List<DAnnotation> deleteList = new ArrayList<>();
-    List<EObject> hideList = new ArrayList<>();
-    for (DAnnotation annotation : list) {
-      EObject element = TitleBlockHelper.getSemanticElementReference(annotation);
-      if(element != null) {
-        DDiagramElement diagElement = DiagramServices.getDiagramServices().getDiagramElement(diagram, element);
-        // case to hide the TB
-        if (diagElement == null) {
-          hideList.add(annotation);
+  private void handleDanglingElementTitleBlocks(DDiagram diagram) {
+    Set<DAnnotation> titleBlocks = getVisibleTitleBlocks(diagram);
+
+    List<DAnnotation> titleBlocksToDelete = new ArrayList<>();
+    List<DAnnotation> titleBlocksToHide = new ArrayList<>();
+
+    for (DAnnotation titleBlock : titleBlocks) {
+      if (TitleBlockHelper.isElementTitleBlock(titleBlock)) {
+        EObject semanticReference = TitleBlockHelper.getSemanticElementReference(titleBlock);
+
+        if (semanticReference != null) {
+          DDiagramElement semanticElementView = DiagramServices.getDiagramServices().getDiagramElement(diagram,
+              semanticReference);
+          if (semanticElementView == null) {
+            titleBlocksToHide.add(titleBlock);
+          }
+
+        } else {
+          titleBlocksToDelete.add(titleBlock);
         }
       }
-      else {
-        // case to delete the TB because it has no reference to a semantic element
-        deleteList.add(annotation);
-      }
     }
 
-    if (!deleteList.isEmpty()) {
-      CapellaServices.getService().removeElements(deleteList);
-      for (DAnnotation annotation : deleteList) {
-        deleteTitleBlock(diagram, annotation);
+    if (!titleBlocksToDelete.isEmpty()) {
+      for (DAnnotation titleBlock : titleBlocksToDelete) {
+        DDiagramElement titleBlockView = DiagramServices.getDiagramServices().getDiagramElement(diagram, titleBlock);

```

Comment:
We dont use the **refresh command** anymore, instead we remove the Element Title Block views **manually**.